lib/compatibility.bzl

```diff
@@ -0,0 +1,171 @@
+"""Skylib module of convenience functions for `target_compatible_with`.
+
+Load the macros as follows in your `BUILD` files:
+```python
+load("@bazel_skylib//lib:compatibility.bzl", "compatibility")
+```
+
+See the [Platform docs](https://bazel.build/docs/platforms#skipping-incompatible-targets) for
+more information.
+"""
+
+load(":selects.bzl", "selects")
+
+def _get_name_from_target_list(targets, joiner=" or "):
+    """Join a list of strings into a string which is suitable as a target name.
+
+    Removes/replaces characters which are not valid as target names.
+
+    Args:
+      target_list: A list of target names.
+      joiner: An optional string to use for joining the list.
+
+    Returns:
+      A string which is a valid target name.
+    """
+    targets_name = joiner.join([s.split(":")[1] for s in targets])
+    name = targets_name.replace("//", "").replace(":", "_").replace("/", "_").replace(".", "_")
+    return name
```

Comment:
You could increase the chance that this name is unique even further by appending something like `str(hash(tuple(targets)))`.

Reply:
Neat idea. I will think about it. Need to run through the logic and clarify the requirements in my head again.

Reply:
Done.